sashimi-webapp/src/database/storage.js

```diff
@@ -0,0 +1,290 @@
+/**
+ *
+ *  CS3283/4 storage.js
+ *  This class acts as a facade for other developers to access to the database.
+ *  The implementation is a non-SQL local storage to support the WebApp.
+ *
+ */
+
+import constants from './constants';
+
+import entitiesCreator from './create/entitiesCreator';
+
+import query from './retrieve/query';
+
+import dataModifier from './data-modifier/dataModifier';
+
+import exceptions from './exceptions';
+
+let databaseName = constants.INDEXEDDB_NAME;
+
+// dummy function to initialize createTables promises
```

Comment:
What is the reason for this dummy promise?

Looks like it is being used in `creationOfTables`

```javascript
function creationOfTables() {
  if (typeof Promise === 'function') {
    return new Promise((resolve, reject) => {
      initCreateTable()
      .then(() =>
        query.isTableExistsInDatabase(constants.ENTITIES_USER)
        .then(isUserTableFirstInitialize =>
          createUserTable(isUserTableFirstInitialize)
          .then(isSuccess => isSuccess)
          .catch(sqlError => reject(sqlError))
          )
        .catch(sqlError => reject(sqlError))
      )
```
How about bypassing that function and use `query.isTableExistsInDatabase` directly?
```javascript
function creationOfTables() {
  if (typeof Promise === 'function') {
    return new Promise((resolve, reject) => {
      query.isTableExistsInDatabase(constants.ENTITIES_USER)
      .then(isUserTableFirstInitialize =>
        createUserTable(isUserTableFirstInitialize)
        .then(isSuccess => isSuccess)
        .catch(sqlError => reject(sqlError))
        )
      .catch(sqlError => reject(sqlError))
      .then(() =>
        query.isTableExistsInDatabase(constants.ENTITIES_ORGANIZATION)
        .then(isOrganizationTableFirstInitialize =>
          createOrganizationTable(isOrganizationTableFirstInitialize))
          .then(isSuccess => isSuccess)
          .catch(sqlError => reject(sqlError))
        .catch(sqlError => reject(sqlError))
      )
```